src/librustdoc/clean/mod.rs

```diff
@@ -1707,12 +1750,122 @@ impl<'a, 'tcx> Clean<Generics> for (&'a ty::Generics,
     }
 }
 
+/// The point of this function is to replace bounds with types.
+///
+/// i.e. `[T, U]` when you have the following bounds: `T: Display, U: Option<T>` will return
```

Comment:
`Option<T>` is not a bound... how does this make sense?

Reply:
The point is to replace T by the type/trait if any. For example:

```
fn<T>(x: T)
where T: Display
```

In there, `T` should be replaced with `Display`. In here, it's all about the rustdoc search, not about finding exact types. If we have two types with the same name, we don't care, we just remove one name from the list and that's it.

Reply:
Sure, but you can get the desired effect without using confusing terminology.

So what you want is "types of arguments" *and* "bounds on parameters in those arguments"?

Why bother with that complication tho? Why not just record all paths in a signature, be it args/return, or bounds/where clauses? Getting all that is less work and much easier to reason about.